cfpb/cfgov-refresh
# related posts refactor/tests

## What is this 
This is a fairly large refactor of Related Posts, motivated by a desire to simplify some of the logic in `def related_posts` (in `v1/models/base.py`) and spiraled into having to update other things, such as `ref.py`. A couple things of note: 
- There is a new field on CFGOVPage -- `related_posts_categories`, derived from a new model `RelatedPostsCategory`. It's the same as `CFGOVPageCategory` in code and behavior. It's meant to be used in place of `specific_categories` on a page with a Related Posts module, but I will follow up with @sarahsimpson09 to see if this is workable.
  - The motivation behind making it a model, as opposed to a list in a Wagtail block, is that it should be easy to update these categories on the back-end, without relying on the publishing flow to see the changes.  Other work in this PR would've necessitated re-publishing many of the pages if not for this migration,  which isn't really an option given that we rely heavily on `date_published` for determining which content to display first or at all.

## How to test
- Pull down the branch, grab the latest production db and import it, run migrations 
- After running migrations, run `python cfgov/manage.py runscript update_categories` to fill in the new `related_posts_categories` with data from `specific_categories`, which we will want to remove eventually
- Check a few pages with related posts modules, e.g. [http://localhost:8000/about-us/blog/what-do-if-youre-wrongfully-billed-medicare-costs/](http://localhost:8000/about-us/blog/what-do-if-youre-wrongfully-billed-medicare-costs/) to confirm they look like what's on www (https://www.consumerfinance.gov/about-us/blog/what-do-if-youre-wrongfully-billed-medicare-costs/)

## TODO
- Update old, failing tests
- Actually write the new tests
